### PR TITLE
Adding debian

### PR DESCRIPTION
**Pull request by nordluf**

### CODE REVIEW

**Comment by insidewhy** [comment]
Out of interest, what happens when you run the centos5 built version on debian?

File: build-release.sh [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-    docker cp $(docker ps -aq | head -n1):/smell-baron/smell-baron .
+    docker build -t buildable-smell-baron -f Dockerfile.alpine $DIR/ || exit 1
     ;;
   *)
     echo "version unsupported, try alpine or centos5"
+    exit 1
     ;;
 esac
+
+docker run --rm buildable-smell-baron cat /smell-baron/smell-baron  > $DIR/smell-baron || exit 1
```
Comment (insidewhy) [inline build-release.sh]: This will clobber the +x fs flag on smell-baron no?

File: build-release.sh [hunk 2]
```diff
@@ -1,21 +1,20 @@
 #!/bin/sh
+DIR=`dirname $(readlink -f $0)`
 
 version=${1:-centos5}
 
 case $version in
   centos5|centos)
-    ln -sf Dockerfile.centos5 Dockerfile
-    docker build -t centos5-smell-baron . || exit 1
```
Comment (insidewhy) [inline build-release.sh:8]: I think `-t $version-smell-baron` might be better, this way the same system can be used to test all versions without each subsequent docker container clobbering the previous one.

**Comment by nordluf** [comment]
Yep, you are absolutely right - made the fix.
I didn't try centos binary in the debian env - I don't want any unexpected issues on production

**Comment by insidewhy** [comment]
It would work fine, the ABI is compatible and the glibc symbols are versioned with the build on Centos using very old API versions. The Centos build should work on any glibc based system from 2.3 and up.

**Comment by insidewhy** [comment]
If you still want this PR accepted in spite of it not being necessary then can you update your first commit WRT my comments and I'll merge it.

**Comment by insidewhy** [comment]
BTW the alpine build is there to support distributions based on musl rather than glibc.

**Comment by nordluf** [comment]
I know about Alpine. I use Alpine and Debian both. 
It is not necessary for me to merge this PR - you are the author, and you know what is better for your project. Just decline this PR :)

**Comment by insidewhy** [comment]
Cool!

**Comment by ohjames** [comment]
The author doesn't always know best, especially when the author is me :)

**Comment by insidewhy** [comment]
There's one advantage to building on debian: taking advantage of any optimisations a more modern gcc than centos 5's might provide. It'll probably be almost inconsequential though, if not totally inconsequential considering how simple `smell-baron` is.

I've cleaned up `build-release.sh` so adding new build systems is trivial, if you feel like porting your commit to that I'll accept it otherwise I'll just cherry-pick your `Docker.alpine`.

**Comment by nordluf** [comment]
I think, that I can create PR with some fixes. Do you want to add something, or - can I make PR from current  code?

**Comment by insidewhy** [comment]
Just need to remove the changes to build-release.sh, the new version picks up the dockerfile based on its extension.

**Comment by insidewhy** [comment]
I've merged your change by cherry-picking your git repository.

**Pull request closed without merging**